vowpalwabbit/options_types.h

```diff
@@ -30,9 +30,9 @@ struct typelist
 
 // is_scalar detection relies on only accepting std::vector<T> for vector types. If that changes, update
 // is_scalar_impl below.
-using supported_options_types =
-    typelist<unsigned int, int, size_t, uint64_t, float, double, char, std::string, bool, std::vector<int>,
-        std::vector<size_t>, std::vector<float>, std::vector<double>, std::vector<char>, std::vector<std::string>>;
+using supported_options_types = typelist<unsigned int, int, size_t, uint64_t, int64_t, float, double, char, std::string,
+    bool, std::vector<int>, std::vector<uint64_t>, std::vector<int64_t>, std::vector<size_t>, std::vector<float>,
```

Comment:
do we also want `std::vector<unsigned int>`?

Reply:
unsigned is already covered by `std::vector<uint64_t>`. Once I've gone through and aligned usage to either `int64_t` or `uint64_t`, then the `unsigned int`, `int` and `size_t` variants will be removed.

The diff is unclear but all this does is add `int64_t`, `std::vector<uint64_t>`, `std::vector<int64_t>`